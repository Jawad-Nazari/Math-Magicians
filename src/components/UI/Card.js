/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ extraclass = '', children }) => (
  <div className={`${extraclass}`}>
    {children}
  </div>
);

Card.propTypes = {
  extraclass: PropTypes.string,
  children: PropTypes.node.isRequired,
};

Card.defaultProps = {
  extraclass: '',
};

export default Card;
