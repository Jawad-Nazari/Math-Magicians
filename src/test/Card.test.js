/* eslint-disable import/no-extraneous-dependencies */
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import renderer from 'react-test-renderer';

import Card from '../components/UI/Card';

describe('Test Card component with snapshot', () => {
  test('should match with snapshot', () => {
    const card = renderer
      .create(
        <Card>
          <h1>Text to pass to test.</h1>
        </Card>,
      )
      .toJSON();

    expect(card).toMatchSnapshot();
  });

  test('should be in the document', () => {
    render(
      <Card>
        <h1>Text to pass to test.</h1>
      </Card>,
    );

    expect(screen.getByText('Text to pass to test.')).toBeInTheDocument();
  });
});
