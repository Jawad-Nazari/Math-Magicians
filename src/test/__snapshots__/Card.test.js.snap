// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Card component with snapshot should match with snapshot 1`] = `
<div
  className=""
>
  <h1>
    Text to pass to test.
  </h1>
</div>
`;
